src/modules/ens/ens.service.unit.spec.ts

```diff
@@ -0,0 +1,256 @@
+    jest.clearAllMocks();
+    module = await Test.createTestingModule({
+      imports: [ConfigModule],
+      providers: [
+        SchedulerRegistry,
+        EnsService,
+        Provider,
+        {
+          provide: OrganizationService,
+          useValue: MockOrgService,
+        },
+        {
+          provide: RoleService,
+          useValue: MockRoleService,
+        },
+        {
+          provide: ApplicationService,
+          useValue: MockApplicationService,
+        },
+        {
+          provide: Logger,
+          useValue: MockLogger,
+        },
+      ],
+    }).compile();
+
+    app = module.createNestApplication();
+    await app.init();
+
+    service = module.get<EnsService>(EnsService);
+  });
+
+  afterEach(async () => {
+    await app.close();
+  });
+
+  describe('Sync ENS', () => {
+    it('syncENS() throw an error', async () => {
+      jest.spyOn(service, 'syncNamespace');
+      jest
+        .spyOn(service, 'getAllNamespaces')
+        .mockResolvedValueOnce(['apps.myorg.iam.ewc']);
+      await service.syncENS();
+      expect(MockLogger.error).toHaveBeenCalledWith(
+        expect.stringContaining(
+          'Error syncing namespace apps.myorg.iam.ewc, owner undefined, Error: unable to parse resolved textData for node:',
+        ),
+      );
+    });
+  });
+
+  describe('ENS Sync Validation', () => {
+    it('syncNamespace() Validate and accept APP Sync Data', async () => {
+      const mockAppServiceSpy = jest.spyOn(
+        MockApplicationService,
+        'handleAppSyncWithEns',
+      );
+      await service.syncNamespace({
+        data: APP_MOCK_DATA,
+        namespace: 'onionapp.apps.onion.iam.ewc',
+        owner: 'onion',
+      });
+      expect(MockLogger.debug).not.toHaveBeenCalled();
+      expect(mockAppServiceSpy).toHaveBeenCalledWith(
+        expect.objectContaining({
+          metadata: APP_MOCK_DATA,
+          namespace: 'onionapp.apps.onion.iam.ewc',
+          owner: 'onion',
+        }),
+      );
+    }, 30000);
+
+    it('syncNamespace() Validate and reject APP Sync Data', async () => {
+      const appSyncData = {
+        data: APP_MOCK_DATA,
+        namespace: 'onionapp.app.onion.iam.ewc',
+        owner: 'onion',
+      };
+
+      const mockAppServiceSpy = jest.spyOn(
+        MockApplicationService,
+        'handleAppSyncWithEns',
+      );
+      await service.syncNamespace(appSyncData);
+      expect(MockLogger.debug).toHaveBeenCalledWith(
+        `Bailed: App with namespace:${appSyncData.namespace} does not have 'apps' subdomain`,
+      );
+      expect(mockAppServiceSpy).not.toHaveBeenCalled();
+    }, 30000);
+
+    it('syncNamespace() Validate and accept ROLE Sync Data with roleType app', async () => {
+      const mockRoleServiceSpy = jest.spyOn(
+        MockRoleService,
+        'handleRoleSyncWithEns',
+      );
+      await service.syncNamespace({
+        data: ROLE_MOCK_DATA,
+        namespace: 'test.roles.onion.iam.ewc',
```

Comment:
Since role type is `app` this role must be in application domain

Reply:
can you be more specific about this? are you saying it should be `test.roles.onion.app.iam.ewc`?

Reply:
I think he is saying that it should be `test.roles.myapp.apps.onion.iam.ewc`